### PR TITLE
Store Read threads

### PR DESCRIPTION
**Pull request by cpbscholten**
### Checklist
- [x] merged database migrations into 1 database migration
- [x] tested database migrations from `origin/staging` (`git checkout staging ; git pull ; bundle exec rails db:reset ; git checkout BRANCH ; bundle exec rails db:migrate`)

### Summary
This PR stores which threads a user has read in the API and adds a boolean to each thread if that user has read the thread.

### Other information
Related UI PR: #217

### CODE REVIEW

File: app/controllers/v1/forum/threads_controller.rb [hunk 1]
```diff
@@ -1,4 +1,14 @@
...
   class ThreadsController < V1::ApplicationController
+    before_action :doorkeeper_authorize!
+    before_action :set_model, only: %i[mark_read]
+
+    def mark_read
+      last = @model.posts.last
+      thread = Forum::ReadThread.where(thread: @model, user: current_user).first_or_create! do |t|
+        t.post = last
+      end
+      thread.update(post: last)
```
Comment (Matthijsy) [inline app/controllers/v1/forum/threads_controller.rb:11]: Isn't this duplicate with the above do block? I think we can remove the do block

Comment (Matthijsy) [reply app/controllers/v1/forum/threads_controller.rb:11]:
```
thread = Forum::ReadThread.first_or_create(user: User.first, thread: Forum::Thread.first)
thread.post = @model.posts.last
thread.save
```

Comment (cpbscholten) [reply app/controllers/v1/forum/threads_controller.rb]: dit is nu aangepast

**Review by Matthijsy — APPROVED** [review]
I think it is fine!

**Pull request merged**